lpc55_isp/src/bin/lpc55_flash.rs

```diff
@@ -333,6 +357,57 @@ fn main() -> Result<()> {
             out.write_all(&m)?;
             println!("CFPA Output written to {:?}", file);
         }
+        ISPCommand::WriteCFPA {
+            update_version,
+            file,
+        } => {
+            do_ping(&mut *port)?;
+
+            let bytes = std::fs::read(&file)?;
+            let mut new_cfpa = lpc55_areas::CFPAPage::from_bytes(
+                bytes[..].try_into().context("CFPA file is not 512 bytes")?,
+            )?;
+
+            // Read the CMPA so we can compare the two to try to avoid locking
+            // the user out of their chip.
+            let m = do_isp_read_memory(&mut *port, 0x9e400, 512)?;
+            let cmpa = lpc55_areas::CMPAPage::from_bytes(m[..].try_into().unwrap())?;
+            if (new_cfpa.dcfg_cc_socu_ns_pin != 0 || new_cfpa.dcfg_cc_socu_ns_dflt != 0)
+                && (cmpa.cc_socu_pin == 0 || cmpa.cc_socu_dflt == 0)
+            {
+                bail!(
+                    "It looks like the CMPA debug settings aren't set but \
+                     the CFPA settings are! This will brick the chip!"
+                );
+                // TODO I guess it's remotely possible that we might want an
+                // override for this check.
+            }
+
+            if update_version {
+                // Read the current CFPA areas to figure out what version we
+                // need to set.
+                let ping = do_isp_read_memory(&mut *port, 0x9_e000, 512)?;
+                let pong = do_isp_read_memory(&mut *port, 0x9_e200, 512)?;
+
+                let ping = lpc55_areas::CFPAPage::from_bytes(ping[..].try_into().unwrap())?;
+                let pong = lpc55_areas::CFPAPage::from_bytes(pong[..].try_into().unwrap())?;
+
+                println!(
+                    "ping sector v={}, pong sector v={}",
+                    ping.version, pong.version
+                );
+                let start_version = u32::max(ping.version, pong.version);
+                if new_cfpa.version != start_version {
+                    new_cfpa.version = start_version;
+                }
+                new_cfpa.update_version();
```

Comment:
The conditional here is confusing, and I don't like that `update_version` hides mutation; maybe just
```suggestion
                new_cfpa.version = start_version + 1;
```

Reply:
I'm open to this, I was attempting to reuse `update_version` since all the other code in here seems to use it. (You are correct that it just hides a mutating increment.)

I have instead removed it.